src/firmware/shared/circular_buffer_test.cpp

```diff
@@ -0,0 +1,175 @@
+extern "C"
+{
+#include "firmware/shared/circular_buffer.h"
+
+#include "shared/constants.h"
+}
+
+#include <gtest/gtest.h>
+
+class CircularBufferTest : public testing::Test
+{
+   protected:
+    virtual void SetUp(void)
+    {
+        circular_buffer = circular_buffer_create(10);
+    }
+
+    CircularBuffer_t* circular_buffer;
+};
+
+TEST_F(CircularBufferTest, circular_buffer_exact_full)
+{
+    float mock_data[]     = {1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0};
+    float expected_data[] = {10.0, 9.0, 8.0, 7.0, 6.0, 5.0, 4.0, 3.0, 2.0, 1.0};
+    int mock_data_size    = sizeof(mock_data) / sizeof(mock_data[0]);
+
+    // Populate circular buffer with mock data
+    for (int i = 0; i < mock_data_size; i++)
+    {
+        circular_buffer_push(circular_buffer, mock_data[i]);
```

Comment:
Would be nice to have  `EXPECT_FALSE(circular_buffer_isFull(circular_buffer));` in one of these loops. Doesn't have to be this one

Reply:
Added to `circular_buffer_front_not_full`